crates/primitives/src/chain_spec.rs

```diff
@@ -107,61 +114,103 @@ impl ChainSpec {
+                ForkKind::TTD(ttd) => ttd.block,
+                _ => None,
+            })
+    }
+
+    /// Get the first block number of the hardfork.
+    pub fn fork_kind(&self, fork: Hardfork) -> Option<ForkKind> {
         self.hardforks.get(&fork).copied()
     }
 
     /// Returns `true` if the given fork is active on the given block
-    pub fn fork_active(&self, fork: Hardfork, current_block: BlockNumber) -> bool {
-        self.fork_block(fork).map(|target| target <= current_block).unwrap_or_default()
+    pub fn fork_active(&self, fork: Hardfork, discriminant: ForkDiscriminant) -> bool {
+        match self.hardforks.get(&fork) {
+            Some(kind) => match kind {
+                ForkKind::Block(block_number) => *block_number <= discriminant.block_number,
+                ForkKind::TTD(ttd) => {
+                    ttd.total_difficulty <= discriminant.total_difficulty &&
+                        ttd.block <= Some(discriminant.block_number)
+                }
+                ForkKind::Time(timestamp) => *timestamp <= discriminant.timestamp,
+            },
+            None => false,
+        }
     }
 
     /// Returns `true` if the DAO fork is supported
     pub fn dao_fork_support(&self) -> bool {
         self.dao_fork_support
     }
 
-    /// Get the Paris status
-    pub fn paris_status(&self) -> ParisStatus {
-        match self.paris_ttd {
-            Some(terminal_total_difficulty) => {
-                ParisStatus::Supported { terminal_total_difficulty, block: self.paris_block }
+    /// The merge terminal total difficulty
+    pub fn terminal_total_difficulty(&self) -> Option<U256> {
+        if let Some(kind) = self.hardforks.get(&Hardfork::MergeNetsplit) {
+            match kind {
+                ForkKind::TTD(ttd) => Some(ttd.total_difficulty),
+                _ => None,
             }
-            None => ParisStatus::NotSupported,
+        } else {
+            None
         }
     }
 
     /// Get an iterator of all harforks with theirs respectives block number
-    pub fn forks_iter(&self) -> impl Iterator<Item = (Hardfork, BlockNumber)> + '_ {
+    pub fn forks_iter(&self) -> impl Iterator<Item = (Hardfork, ForkKind)> + '_ {
         self.hardforks.iter().map(|(f, b)| (*f, *b))
     }
 
-    /// Creates a [`ForkFilter`](crate::ForkFilter) for the given [BlockNumber].
-    pub fn fork_filter(&self, block: BlockNumber) -> ForkFilter {
-        let future_forks =
-            self.forks_iter().map(|(_, b)| b).filter(|b| *b > block).collect::<Vec<_>>();
+    /// Creates a [`ForkFilter`](crate::ForkFilter) for the given [ForkDiscriminant].
+    pub fn fork_filter(&self, discriminant: ForkDiscriminant) -> ForkFilter {
+        let future_forks = self
+            .forks_iter()
+            .filter(|(f, _)| !self.fork_active(*f, discriminant))
+            .filter_map(|(_, k)| match k {
+                ForkKind::Block(block_number) => Some(block_number),
+                ForkKind::TTD(_) => None,
```

Comment:
Is the merge not part of the fork filter?

Reply:
I don't think so, like on forkid. Mayby @Rjected can confirm?

Reply:
See here:
https://github.com/ethereum/go-ethereum/blob/59a48e0289b1a7470a8285e665cab12b29117a70/core/forkid/forkid.go#L76-L98

Which only gathers forks by timestamp and block num, and the merge is not one of those forks

Reply:
forkfilter in geth using the same `gatherForks` method here:
https://github.com/ethereum/go-ethereum/blob/59a48e0289b1a7470a8285e665cab12b29117a70/core/forkid/forkid.go#L138